controllers/wifi/wifi.controller.js

```diff
@@ -1,62 +1,89 @@
-import { RoomBooking, WifiDb } from '../../models/associations.js';
+import { FlatBooking, RoomBooking, WifiDb } from '../../models/associations.js';
 import {
   ROOM_STATUS_CHECKEDIN,
   STATUS_ACTIVE,
   STATUS_INACTIVE
 } from '../../config/constants.js';
 import APIError from '../../utils/ApiError.js';
 import Sequelize from 'sequelize';
+import database from '../../config/database.js';
 import moment from 'moment';
 
 export const generatePassword = async (req, res) => {
-  const isCheckedin = await RoomBooking.findOne({
+  const t = await database.transaction();
+  req.transaction = t;
+
+  const isRoomCheckedin = await RoomBooking.findOne({
+    where: {
+      cardno: req.user.cardno,
+      checkout: { [Sequelize.Op.gte]: moment().format('YYYY-MM-DD') },
+      status: ROOM_STATUS_CHECKEDIN
+    }
+  });
+
+  const isFlatCheckedin = await FlatBooking.findOne({
     where: {
       cardno: req.user.cardno,
       checkout: { [Sequelize.Op.gte]: moment().format('YYYY-MM-DD') },
       status: ROOM_STATUS_CHECKEDIN
     }
   });
-  if (!isCheckedin) {
+
+  if (!isRoomCheckedin && !isFlatCheckedin) {
     throw new APIError(401, 'User not checkedin');
   }
 
   const count = await WifiDb.count({
     where: {
       cardno: req.user.cardno,
       status: STATUS_INACTIVE,
-      roombookingid: isCheckedin.bookingid
+      roombookingid: isRoomCheckedin?.bookingid || isFlatCheckedin?.bookingid
     }
   });
   if (count < 5) {
+    const roombookingid =
+      isRoomCheckedin?.bookingid || isFlatCheckedin?.bookingid;
+
+    if (!roombookingid) {
+      throw new APIError(400, 'Booking ID not found');
+    }
```

Comment:
this will never happen right, as you're already checking this on line 32
```
  if (!isRoomCheckedin && !isFlatCheckedin) {
    throw new APIError(401, 'User not checkedin');
  }
```